pkg/recommendation/device.go

```diff
@@ -113,6 +113,7 @@ func (r *cStorPoolClusterRecommendationRequest) GetRecommendation() map[string]t
 
 		cStorPoolClusterRecommendationValue := nodeCapacityBlockDeviceMap.getDeviceRecommendation(r.Request.Spec.PoolCapacity, r.Request.Spec.DataConfig)
 		cStorPoolClusterRecommendationValue.RequestSpec = r.Request.Spec
+		cStorPoolClusterRecommendationValue.ObjectMeta = r.Request.ObjectMeta
```

Comment:
@slalwani97  we need name and namespace only there may be uid or other stuff let's not assign the full object

Reply:
Its the same thing shovan, other fields will have the default value in both cases.

Reply:
The labels of the request will be copied to recommendation if we use this. If we want to create cr later.